src/kernel/bpf/prog.rs

```diff
@@ -0,0 +1,166 @@
+use std::borrow::Cow;
+use std::fmt::Debug;
+use std::fmt::Display;
+use std::fmt::Formatter;
+use std::fmt::Result as FmtResult;
+use std::mem::size_of;
+use std::str::FromStr;
+
+use crate::inspect::SymInfo;
+use crate::symbolize::FindSymOpts;
+use crate::symbolize::ResolvedSym;
+use crate::symbolize::SrcLang;
+use crate::Addr;
+use crate::Error;
+use crate::Result;
+use crate::SymType;
+
+/// BPF kernel programs show up with this prefix followed by a tag and
+/// some other meta-data.
+const BPF_PROG_PREFIX: &str = "bpf_prog_";
+
+
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+#[repr(transparent)]
+pub struct BpfTag([u8; 8]);
+
+impl Display for BpfTag {
+    fn fmt(&self, f: &mut Formatter<'_>) -> FmtResult {
+        for b in self.0 {
+            let () = write!(f, "{b:02x}")?;
+        }
+        Ok(())
+    }
+}
+
+impl From<[u8; 8]> for BpfTag {
+    fn from(other: [u8; 8]) -> Self {
+        Self(other)
+    }
+}
+
+impl FromStr for BpfTag {
+    type Err = ();
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if s.as_bytes().len() != 2 * size_of::<BpfTag>() {
+            return Err(())
+        }
+
+        let mut tag = [0; size_of::<BpfTag>()];
+        (0..s.len())
+            .step_by(2)
+            .enumerate()
+            .try_for_each(|(i, idx)| {
+                tag[i] = u8::from_str_radix(&s[idx..idx + 2], 16).map_err(|_| ())?;
+                Ok(())
+            })?;
+
+        Ok(Self(tag))
+    }
+}
+
+
+/// Information about a BPF program.
+#[derive(Debug)]
+pub struct BpfProg {
+    addr: Addr,
+    name: Box<str>,
+    tag: BpfTag,
+}
+
+impl BpfProg {
+    /// Parse information about a BPF program from part of a `kallsyms`
+    /// line.
+    pub fn parse(s: &str, addr: Addr) -> Option<Self> {
+        let s = s.strip_prefix(BPF_PROG_PREFIX)?;
+        let (tag, name) = s.split_once('_')?;
```

Comment:
judging by https://doc.rust-lang.org/std/string/struct.String.html#examples-80, it will fail for `bpf_prog_07df9015606e4140` case (i.e., when there is no program name in the symbol). is that ok?

Reply:
Changed behavior to support empty name.